src/solvers/particle_swarm.jl

```diff
@@ -259,35 +259,47 @@ function particle_swarm!(
         inverse_retraction_method=inverse_retraction_method,
         vector_transport_method=vector_transport_method,
     )
-    o = decorate_state!(o; kwargs...)
+    o = decorate_state!(pss; kwargs...)
     return get_solver_return(solve!(mp, o))
 end
 
 #
 # Solver functions
 #
 function initialize_solver!(mp::AbstractManoptProblem, s::ParticleSwarmState)
+    M = get_manifold(mp)
     j = argmin([get_cost(mp, p) for p in s.x])
-    return s.g = deepcopy(s.x[j])
+    s.g = copy(M, s.x[j])
+    return s
 end
 function step_solver!(mp::AbstractManoptProblem, s::ParticleSwarmState, ::Any)
     M = get_manifold(mp)
+    # Allocate two tangent vectors
+    cognitive_infl = zero_vector(M, first(s.p))
+    social_infl = zero_vector(M, first(s.p))
+    xOld = copy(M, first(s.p))
     for i in 1:length(s.x)
-        s.velocity[i] =
+        inverse_retract!(M, cognitive_infl, s.x[i], s.p[i], s.inverse_retraction_method)
+        inverse_retract!(M, social_infl, s.x[i], s.g, s.inverse_retraction_method)
+        # add v = inertia * v + cw*cog_infl + sw*soc_infl
+        # where the last two are randomly shortened a bit
+        copyto!(
```

Comment:
This should be a bit faster using broadcasting, like
```julia
s.velocity[i] .= s.inertia .* s.velocity[i] .+ s.cognitive_weight .* rand(1) .* cognitive_infl .+ s.social_weight .* rand(1) .* social_infl
```

Reply:
Oh, I actually _never_ know which one is faster here; because I always have FixedRankManifolds in mind where the tangent vector is a struct and then I do not know whether `.=` works. Are you sure it works?

Reply:
I've spent quite some time ensuring that broadcasting works for all our tangent vectors, including the ones defined by structs. If it doesn't in some case then let me know and I will take a look.

Broadcasting is almost always faster than `copyto!`.

Reply:
ok, will change that as well. First the 3 new fields and a bit of allocations / type inference when creating a state.